express/index.js

```diff
@@ -27,35 +27,45 @@ const express = require('express');
  */
 require('express-async-errors');
 const http = require('http');
+const http2 = require('spdy');
 const https = require('https');
 const pem = require('pem');
 
 const app = express();
-const { PORT = 3000, HOST = 'localhost', SSL, CLUSTER } = process.env;
+const { PORT = 3000, HOST = 'localhost', SSL, CLUSTER, HTTP2 } = process.env;
 const isHttps = SSL === '1' ? true : false;
+const isHttp2 = HTTP2 === 'true' ? true : false;
 require('./app').setup(app);
 
 const listener = function listener() {
   const { address, port } = this.address();
-  const protocol = isHttps ? 'https' : 'http';
+  const protocol = (isHttps || isHttp2) ? 'https' : 'http';
   const stop = Date.now();
   /* eslint-disable no-console */
   console.log(`startup time: ${stop - start}`);
   console.log('Server listening on %s://%s:%d', protocol, address, port);
 };
 
 function createServer() {
-  /* Start Server based on protocol */
-  isHttps
-    ? pem.createCertificate({ days: 1, selfSigned: true }, (err, keys) => {
-        if (err) {
-          throw err;
-        }
-        https
-          .createServer({ key: keys.serviceKey, cert: keys.certificate }, app)
-          .listen(PORT, HOST, listener);
-      })
-    : http.createServer(app).listen(PORT, HOST, listener);
+  pem.createCertificate({ days : 1, selfSigned : true }, (err, keys) => {
```

Comment:
Not sure it matters a ton, but we don't need to create a certificate if using HTTP. Same goes for the logic in koa's entry point.